test/object-store/migrations.cpp

```diff
@@ -577,29 +577,267 @@ TEST_CASE("migration: Automatic") {
+            Schema schema = {
+                {"child_table",
+                    {
+                        {"value", PropertyType::Int},
+                    }},
+                {"parent_table",
+                    {
+                        {"child_property", PropertyType::Object | PropertyType::Nullable, "child_table"},
+                    }},
+            };
+            auto realm = Realm::get_shared_realm(config);
+            realm->update_schema(schema, 1);
+            realm->begin_transaction();
+            auto child_table = ObjectStore::table_for_object_type(realm->read_group(), "child_table");
+            Obj child_object = child_table->create_object();
+            child_object.set("value", 42);
+            auto parent_table = ObjectStore::table_for_object_type(realm->read_group(), "parent_table");
+            auto child_object_key = child_object.get_key();
+            parent_table->create_object().set_all(child_object_key);
+            parent_table->create_object().set_all(child_object_key);
+            realm->commit_transaction();
+            REQUIRE(parent_table->size() == 2);
+            REQUIRE(child_table->size() == 1);
+            
+            REQUIRE_NOTHROW(realm->update_schema(
+                set_embedded(schema, "child_table", true), 2, [](auto old_realm, auto new_realm, auto&) {
+                
+                Object parent_object1(new_realm, "parent_table", 0);
+                CppContext context(new_realm);
+                Object child_object1 =
+                any_cast<Object>(parent_object1.get_property_value<util::Any>(context, "child_property"));
+                int value = any_cast<Int>(child_object1.get_property_value<util::Any>(context, "value"));
+                
+                auto child_table = ObjectStore::table_for_object_type(new_realm->read_group(), "child_table");
+                Obj child_object2 = child_table->create_object();
+                child_object2.set("value", value);
+                
+                Object parent_object2(new_realm, "parent_table", 1);
+                parent_object2.set_property_value(context, "child_property", util::Any(child_object2));
+            }));
+            
+            REQUIRE(realm->schema_version() == 2);
+            REQUIRE(parent_table->size() == 2);
+            REQUIRE(child_table->size() == 2);
+            for (int i = 0; i < 2; i++) {
+                Object parent_object(realm, "parent_table", i);
+                CppContext context(realm);
+                Object child_object =
+                any_cast<Object>(parent_object.get_property_value<util::Any>(context, "child_property"));
+                int value = any_cast<Int>(child_object.get_property_value<util::Any>(context, "value"));
+                REQUIRE(value == 42);
+            }
+        }
+        
+        SECTION("change table to embedded - adding more links in migration block") {
```

Comment:
really great to see all these cases tested 💯

Reply:
Thanks. :)